packages/extension-ui/src/Popup/RestoreJson.tsx

```diff
@@ -0,0 +1,108 @@
+// Copyright 2019-2020 @polkadot/extension-ui authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { KeyringPair$Json } from '@polkadot/keyring/types';
+import React, { useCallback, useState } from 'react';
+import { InputWithLabel, InputFileWithLabel, Button, Address } from '../components';
+import { u8aToString } from '@polkadot/util';
+import styled from 'styled-components';
+import { jsonRestore, jsonVerifyPassword, jsonVerifyFile } from '../messaging';
+
+import { Header } from '../partials';
+
+type Props = {};
+
+interface FileState {
+  address: string | null;
+  isFileValid: boolean;
+  json: KeyringPair$Json | null;
+}
+
+interface PassState {
+  isPassValid: boolean;
+  password: string;
+}
+
+const acceptedFormats = ['application/json', 'text/plain'].join(', ');
+
+async function parseFile (file: Uint8Array): Promise<FileState> {
+  try {
+    const json = JSON.parse(u8aToString(file));
```

Comment:
```suggestion
    const json = JSON.parse(u8aToString(file)) as KeyringPair$Json;
```

We need to explicitly cast, since the parsing output is untyped. (Basically saying, he, this is what I expect)

Reply:
(Also fixes the linting errors below, since the type is now known)